### DISCUSSION TITLE
How to have a different number of posts on the first blog page than on the following ones?

### DISCUSSION THREAD

**Discussion by pftnhr** [datenstrom/community#899] [Ask a question] [answered]
I have some content on the first page of the blog, so I want to have one less post there than on the following pages. But since `$pages` and the limit are declared before my if statement, it seems that I cannot give it another `blogPaginationLimit`. 

Here’s the code from my blog-start.html:

``` php
<?php $pages = $this->yellow->page->getPages("blog")->paginate($this->yellow->system->get("blogPaginationLimit")) ?>

<?php if ($pages->isPagination() && $pages->getPaginationNumber() == "1") : ?>	
	<h1><?php echo $this->yellow->page->getHtml("titleContent") ?></h1>
	<?php echo $this->yellow->page->getContentHtml() ?>
<?php endif ?>
```

Please, can someone give me a hint on how to solve my problem?

For search purposes: pagination, limit

**Comment by pftnhr** [comment]
I now have the idea of prepending the content that comes before my regular posts on the first blog page using `pages->prepend($page)`. Unfortunately, I can't read the API documentation in a way that I really understand what I have to do. So I tried the following:

``` php
<?php if ($pages->isPagination() && $pages->getPaginationNumber() == "1") : $pages->prepend("shared")?>
```

but than Yellow throws an error:

```
Fatal error: Uncaught Error: Call to a member function set() on string in /www/htdocs/w01e3007/frittiert.es/system/layouts/frittiertes-blog-start.html:30
```

It is interesting that `$pages->prepend("shared")` does not seem to cause a direct error, only in the follow-up. Unfortunately, I don't understand the connection. Can someone please enlighten me?

**Reply by pftnhr** [reply, marked as answer]
After a while of trial and error, I came to the conclusion that it had to work like this:
``` php
<?php $pages = $this->yellow->page->getPages("blog")->prepend($this->yellow->page->getPage("intro"))->paginate($this->yellow->system->get("BlogPaginationLimit"))->remove($this->yellow->page->getPage("intro")) ?>
```
Fiddlesticks. Mark told me in the Discord chat,
```
class YellowPageCollection extends ArrayObject
https://www.php.net/manual/en/class.arrayobject.php
There you can see that PHP returns void. This was okay to change until PHP 8 come around
``` 
and told me to use a workaround and write the code line by line in the layout, which I did. From then on, it worked the way I wanted it to.

And now here is the working code in my `blog-start.html`:
``` php
<?php $pages = $this->yellow->page->getPages("blog"); ?>
<?php $pages->prepend($this->yellow->page->getPage("intro")); ?>
<?php $pages->paginate($this->yellow->system->get("BlogPaginationLimit"))->remove($this->yellow->page->getPage("intro")); ?>
```
This will result in one less post on the blog homepage than on the following blog pages. With this code, I then insert content from `/1-home/page.md` on my blog home page, which I also display differently from the blog posts:
``` php
<?php if ($pages->isPagination() && $pages->getPaginationNumber() == "1") : ?>
	<h1><?php echo $this->yellow->page->getHtml("titleContent") ?></h1>
	<?php echo $this->yellow->page->getContentHtml() ?>
<?php endif ?>
```

All thanks to the Yellow Community and especially to Mark!

**Reply by markseuffert** [reply]
My choice of words was less diplomatic in the Discord chat. Inheritance from ArrayObject doesn't make sense anymore, if you want to chain methods [[1]](https://en.wikipedia.org/wiki/Method_chaining) [[2]](https://medium.com/@hendurhance/getting-started-with-method-chaining-in-object-oriented-programming-b3fd60bffe1e), but the return type of methods has to match the built-in methods [[3]](https://php.watch/versions/8.1/internal-method-return-types). This is something to discuss in the next API review. Thank you Robert for sharing working code, much appreciated.
